src/main/java/org/truffleruby/core/exception/CoreExceptions.java

```diff
@@ -845,6 +845,13 @@ public DynamicObject encodingCompatibilityErrorIncompatibleWithOperation(Encodin
         return encodingCompatibilityError(StringUtils.format("incompatible encoding with this operation: %s", encoding), currentNode);
     }
 
+    @TruffleBoundary
+    public DynamicObject encodingError(String message, Node currentNode) {
+        DynamicObject exceptionClass = context.getCoreLibrary().getEncodingErrorClass();
+        DynamicObject errorMessage = StringOperations.createString(context, StringOperations.encodeRope(message, UTF8Encoding.INSTANCE));
+        return ExceptionOperations.createRubyException(context, exceptionClass, errorMessage, currentNode, null);
+    }
```

Comment:
This should be above the other 2 methods, since it's a different error class.

Reply:
Fixed.